esp-hal/src/timer/systimer.rs

```diff
@@ -49,7 +49,7 @@
 //!
 //! let mut systimer = SystemTimer::new(peripherals.SYSTIMER);
 //!
-//! // Get the current timestamp, in microseconds:
+//! // Get the current timestamp, in ticks of 16MHz (1/16 us):
```

Comment:
I don't think we want to meantion the actual value, as I don't think it's correct for the S2 or for different chrystals. We should be more clear that the actual tick rate can be accessed via `SystemTimer::ticks_per_second()`.

Reply:
The point I have with this, is that the value is not "microseconds". Can change the text to be vaguer.

I noticed this when trying different approaches for `defmt` timing output. Ended up using `esp_hal::time::now` (which I think uses this one, underneath, but has wider MCU compatibility).

Reply:
I didn't say you are mistaken to change this :) Just that the result isn't perfect either.

> I noticed this when trying different approaches for defmt timing output.

Have you considered enabling [embassy-time/defmt-timestamp-uptime-s](https://docs.rs/crate/embassy-time/latest/features)? Or am I misunderstanding what you're trying to do?

Reply:
Had a fresh look at the text in its context. How about:

```
Get the current tick count:
```

I think using `timestamp` makes one think in terms of `us`. There's a mention to `SystemTimer::ticks_per_second` just a few lines below, so I think this.. reduction in text would work sufficiently great.

>Have you considered enabling [embassy-time/defmt-timestamp-uptime-s](https://docs.rs/crate/embassy-time/latest/features)?

I haven't. I am on my way to Embassy, but not there yet. Will check.

Reply:
> Get the current tick count:

That sounds good to me

Reply:
Changed; let me know if I am expected to do something more on this.